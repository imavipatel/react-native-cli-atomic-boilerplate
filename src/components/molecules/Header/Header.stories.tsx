import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';
import { Header } from './Header';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

export const actions = {
  onClick: action('onClick'),
};

storiesOf('Header', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => <Header>Header Text</Header>);
