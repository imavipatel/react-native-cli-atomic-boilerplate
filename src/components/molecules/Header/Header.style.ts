import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  headerContainer: {
    minWidth: 200,
    minHeight: 45,
    justifyContent: 'center',
    alignItems: 'center',
  },
  headerText: {
    color: 'black',
    fontWeight: '600',
    fontSize: 30,
  },
});
