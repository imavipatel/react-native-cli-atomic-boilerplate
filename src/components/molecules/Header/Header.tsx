import React, { ReactNode } from 'react';
import { Text, View } from 'react-native';
import { styles } from './Header.style';

export interface HeaderProps {
  children: ReactNode;
}

export const Header: React.FC<HeaderProps> = props => {
  return (
    <View style={styles.headerContainer} testID="header">
      <Text style={styles.headerText} testID="header-text">
        {props.children}
      </Text>
    </View>
  );
};
