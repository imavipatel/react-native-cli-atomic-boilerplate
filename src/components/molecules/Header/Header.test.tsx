import React from 'react';
import { render } from '@testing-library/react-native';
import { Header } from './Header';

describe('Header', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(<Header>Header Text</Header>);

    expect(tree).toMatchSnapshot();
  });
});
