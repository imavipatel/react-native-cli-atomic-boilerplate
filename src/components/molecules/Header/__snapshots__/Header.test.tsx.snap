// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "minHeight": 45,
      "minWidth": 200,
    }
  }
  testID="header"
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 30,
        "fontWeight": "600",
      }
    }
    testID="header-text"
  >
    Header Text
  </Text>
</View>
`;
